File: Android/app/src/main/java/br/com/desafioandroid/wsconsumer/RetrofitConfig.java
package br.com.desafioandroid.wsconsumer;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class RetrofitConfig {

    private final Retrofit retrofit;

    public static final String preferencias = "preferencias";
    public static final String objOffline = "objOffline";

    public RetrofitConfig() {

        retrofit = new Retrofit.Builder()
                .baseUrl("https://api.github.com/")
                .addConverterFactory(GsonConverterFactory.create())
                .build();
    }

    public ApiService getService() {
        return RetrofitConfig.this.retrofit.create(ApiService.class);
    }

}

File: app/src/main/java/br/com/eriberto/desafioandroidconcrete/model/retrofit/RetrofitConfig.kt
package br.com.eriberto.desafioandroidconcrete.model.retrofit

import br.com.eriberto.desafioandroidconcrete.model.retrofit.services.ForksService
import br.com.eriberto.desafioandroidconcrete.model.retrofit.services.RepositoriosService
import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import retrofit2.Retrofit
import retrofit2.adapter.rxjava.RxJavaCallAdapterFactory
import retrofit2.converter.gson.GsonConverterFactory

class RetrofitConfig {
    var interceptor : HttpLoggingInterceptor = HttpLoggingInterceptor().setLevel(HttpLoggingInterceptor.Level.BODY)
    var client = OkHttpClient.Builder().addInterceptor(interceptor).build()!!

    private val retrofit = Retrofit.Builder()
        .baseUrl("https://api.github.com/")
        .client(client)
        .addCallAdapterFactory(RxJavaCallAdapterFactory.create())
        .addConverterFactory(GsonConverterFactory.create())
        .build()

    fun repositoriosService() = retrofit.create(RepositoriosService::class.java)
    fun forksService() = retrofit.create(ForksService::class.java)
}